### ISSUE TITLE
Re-implement the old data-harvesting APIs

### ISSUE THREAD

**Issue by jechols** [open-oni/open-oni#302]
<del>The IIIF Presentation JSON responses don't provide the same level of information that was previously available via the batch/issue/title APIs.  Given that, it may be a mistake to have IIIF *replace* those old APIs.  I recommend reinstating the old APIs and making the IIIF stuff live at different URLs.  The IIIF presentation document has a recommendation for how to structure these URLs, so it may be better to just match that structure anyway.</del>

To summarize the ramblings that came below, our final plan was "...simply reimplementing the old APIs (or something like them) at a different endpoint. We can't really change the existing URLs now that we have other adopters, but Scoop will need a way to get at bulk data efficiently."

**Comment by jechols** [comment]
No way we'll have time for this for 1.0, so maybe it's a 2.0 issue.  Changing the API endpoints in any 1.x version would be a major pain for end users.

**Comment by jechols** [comment]
This is now an issue for somebody other than myself (NC mentioned having to rewrite things to use the new APIs)

Given that it would add new compatibility issues, though, it may make more sense to just put the old endpoints (a) in a plugin, or (b) at a new `/api` endpoint or something

**Comment by jechols** [comment]
This recently got trickier.  IIIF 3.0 dropped the recommended URI patterns (https://github.com/IIIF/api/issues/1214), and says not to change URIs *ever*.

So we have to pick the lesser of two evils:

- Reimplement the LC APIs at a different endpoint, making people have to rewrite tools that work with LC to work differently with ONI.  (And possibly worse is the fact that tools *will* get a response; it just won't work with those LC-compatible tools)
- Break the IIIF URLs so they live under something a lot more sane, like `/iiif/title/manifest.json` or something.  People using LC-compatible tools don't get hosed, but people using IIIF do.

**Comment by jechols** [comment]
Final solution will have to be simply reimplementing the old APIs (or something like them) at a different endpoint. We can't really change the existing URLs now that we have other adopters, but Scoop will need a way to get at bulk data efficiently.

**Comment by jechols** [comment]
Giving this ticket a new title to indicate the current plan (simply re-implementing the old APIs) since moving the IIIF endpoints is no longer a goal.

**Comment by techgique** [comment]
Title, Issue, etc model data via JSON endpoint. UOregon example reimplemented endpoints below:
- https://github.com/uoregon-libraries/oregon-oni/blob/main/json_api.py
- https://github.com/uoregon-libraries/oregon-oni/blob/main/urls.py

**Comment by jechols** [comment]
Per issue #602 and slack conversations as well, I forgot about DRF being useful for bulk-data APIs to allow throttling. Also would be useful for IIIF for the same reason. If possible DRF + read-only endpoints would be in core, but not worth undue research/dev time if it's easier to do a plugin for the data-harvesting endpoints in order to simplify the destructive endpoints we have planned.

**Comment by techgique** [comment]
Work continuing on this process: https://github.com/open-oni/open-oni/tree/feature/core-api

Have basic version of all endpoints from [LoC / ChronAm API](https://chroniclingamerica.loc.gov/about/api/#json-views) implemented. Took longer than I thought once I'd gotten over the initial DRF learning curve. Has still been additional work adapting the original code into how DRF works.

Remaining tasks before PR ready:
- Pagination
- 404 responses for param values not
- Testing
- Documentation update
- Update robots.txt?

Optional task with extra time:
- Rework newspapers.json data structure and code
